query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala

```diff
@@ -64,14 +65,21 @@ final case class AggregateMapReduce(aggrOp: AggregationOperator,
             sourceSchema: ResultSchema): Observable[RangeVector] = {
     val valColType = RangeVectorTransformer.valueColumnType(sourceSchema)
     val aggregator = RowAggregator(aggrOp, aggrParams, valColType)
+
     def grouping(rv: RangeVector): RangeVectorKey = {
       val groupBy: Map[ZeroCopyUTF8String, ZeroCopyUTF8String] =
         if (by.nonEmpty) rv.key.labelValues.filter(lv => byLabels.contains(lv._1))
         else if (without.nonEmpty) rv.key.labelValues.filterNot(lv =>withoutLabels.contains(lv._1))
         else Map.empty
       CustomRangeVectorKey(groupBy)
     }
-    RangeVectorAggregator.mapReduce(aggregator, skipMapPhase = false, source, grouping)
+
+    // IF no grouping is done AND prev transformer is Periodic (has fixed length), use optimal path
+    sourceSchema.fixedVectorLen.filter { _ => without.isEmpty && by.isEmpty }.map { fixedLen =>
```

Comment:
Doing that would require more work and you would need more memory for intermediate group results too, so benefits are less clear.  So that's left to another PR.